### PR TITLE
allow early return outside of main

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
Fixes: #9792

When evaluating file, we need to allow early return if we evaluate script file first.

### CODE REVIEW

File: tests/fixtures/formats/early_return_outside_main.nu [hunk 1]
```diff
@@ -0,0 +1,11 @@
+let y = 'not hiya'
+
+return
+
+def test [] {
+    error make {msg: "this should not show"}
+}
```
Comment (amtoine) [inline tests/fixtures/formats/early_return_outside_main.nu:7]: is that relevant to the test?

i could check the fix without that part, so i'm just wondering if i'm missing something :relieved:

**Review by amtoine — COMMENTED** [review]
i'm liking this :pray: 

just a minor question on my end to make sure i understand the tests :relieved:

File: tests/fixtures/formats/early_return_outside_main.nu (hunk 1 above)
Comment (WindSoilder) [reply tests/fixtures/formats/early_return_outside_main.nu]: Oh, I forget to remove `test` function, thanks!

**Review by amtoine — APPROVED** [review]
looks good to me, thanks for the test @WindSoilder :pray:
